framework/src/main/java/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -135,7 +136,12 @@ public DefaultTypeArgumentInference(AnnotatedTypeFactory typeFactory) {
 
         final Set<TypeVariable> targets = TypeArgInferenceUtil.methodTypeToTargets(methodType);
 
-        if (assignedTo == null && TreeUtils.getAssignmentContext(pathToExpression) != null) {
+        if ((pathToExpression.getParentPath() != null
+                        && pathToExpression.getParentPath().getLeaf() != null
+                        && pathToExpression.getParentPath().getLeaf().getKind()
```

Comment:
I'm wondering whether using `TreeUtils.enclosingMethodOrLambda` and then checking for `LAMBDA_EXPRESSION` would be a bit shorter.

Reply:
The type arguments cannot be inferred if the immediate parent is lambda expression.  `TreeUtils.enclosingMethodOrLambda` might return a lambda expression that is the parent's parent, so it doesn't work.

Reply:
Couldn't the method invocation be parenthesized?
Couldn't one write `mapper = dsData1 -> ((ImmutableMap.of(...)))`? Not the nicest style, but would be a bit odd not to handle.